nixos/modules/services/monitoring/prometheus/exporters.nix

```diff
@@ -295,6 +296,12 @@ in
         Please specify either 'services.prometheus.exporters.mail.configuration'
           or 'services.prometheus.exporters.mail.configFile'.
       '';
+    } {
+      assertion = cfg.mysqld.runAsLocalSuperUser -> config.services.mysql.enable;
+      message = ''
+        The exporter is configured to run as 'services.mysql.user', but
+          'services.mysql.enable' is set to false.
```

Comment:
We can't use it remotely?

Reply:
The intention behind this was that you'd very likely want to run the exporter as `DynamicUser` unless you'd like to listen on a local mysql socket, in which case it makes sense to run it under the `mysql` system user for permission reasons. If you're going to listen to a remote mysql service anyway, it makes little sense to set up the `mysql` user only for this service, and I would assume it to be a result of misconfiguration.